### PR TITLE
Support JeOS on VMware

### PR DESCRIPTION
**Pull request by mnowaksuse**
This adds support for running JeOS on VMware (net install won't work as
VMware does not support direct kernel boot).

Serial port is exposed on ESX server as `virsh console' is not supported
by esx libvirt driver, netcat is used instead.

### CODE REVIEW

**Comment by mnowaksuse** [comment]
Requires https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/1300.

File: consoles/sshVirtsh.pm [hunk 1]
```diff
@@ -315,6 +332,26 @@ sub add_disk {
 sub define_and_start {
     my ($self) = @_;
 
+    my $remote_vmm = "";
+    if ($self->vmm_family eq 'vmware') {
+        my $libvirtauthfilename = '/var/lib/libvirt/images/libvirt.auth';
```
Comment (coolo) [inline consoles/sshVirtsh.pm]: does it have to live in /images ?

File: consoles/sshVirtsh.pm [hunk 2]
```diff
@@ -327,18 +364,18 @@ sub define_and_start {
     $chan->close();
 
     # shut down possibly running previous test (just to be sure) - ignore errors
-    $self->{ssh}->channel()->exec("virsh destroy " . $self->name);
-    $self->{ssh}->channel()->exec("virsh undefine " . $self->name);
+    $self->{ssh}->channel()->exec("virsh " . "$remote_vmm" . " destroy " . $self->name);
```
Comment (coolo) [inline consoles/sshVirtsh.pm]: just put the variable in the string - no need to concat if it's a string (while $self->name is a function call) and I find "$virsh $remote_vmm destroy " easier to read

File: consoles/sshVirtsh.pm (hunk 1 above)
Comment (mnowaksuse) [inline consoles/sshVirtsh.pm]: Nope. `/var/lib/libvirt/images/` is just our favorite dumpster folder, see `$xmlfilename`. Suggestions?

**Comment by mnowaksuse** [comment]
z/KVM test got updated in os-autoinst/os-autoinst-distri-opensuse#1300.

File: consoles/sshVirtsh.pm (hunk 1 above)
Comment (coolo) [inline consoles/sshVirtsh.pm]: I would have expected a temporary file actually

**Pull request merged**